src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -835,28 +836,58 @@ export default class Fms {
      * @param routeString {string}
      * @return {boolean}
      */
-    replaceDepartureProcedure(routeString) {
-        const routeStringElements = routeString.split(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
+    replaceDepartureProcedure(routeString, airportIcao) {
+        const routeStringElements = routeString.toUpperCase().split(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
 
-        if (routeStringElements.length !== 3) {
+        if (routeStringElements.length > 3) {
             return [false, 'departure procedure format not understood'];
         }
 
-        const procedureId = routeStringElements[1];
+        let procedureId = routeStringElements[0];
 
-        if (!NavigationLibrary.hasProcedure(procedureId)) {
+        if (routeStringElements.length === 3) {
+            procedureId = routeStringElements[1];
+        }
+
+        const sidModel = NavigationLibrary.getProcedure(procedureId);
+
+        if (_isNil(sidModel)) {
             return [false, `unknown procedure "${procedureId}"`];
         }
 
-        const wasSuccessful = this._routeModel.replaceDepartureProcedure(routeString);
+        if (routeStringElements.length === 1) { // RouteString looks like PROC
+            const exitPoint = _findLast(this.waypoints, (waypointModel) => sidModel.hasExit(waypointModel.name));
 
-        if (wasSuccessful) {
-            this._updateDepartureRunwayFromRoute();
+            if (!exitPoint) {
+                return [false, `the ${procedureId.toUpperCase()} departure doesn't have an exit along our route`];
+            }
 
-            return [true, ''];
+            routeStringElements.push(exitPoint.name);
         }
 
-        return [false, `route of "${routeString}" is not valid`];
+        if (routeStringElements.length === 2) { // RouteString looks like PROC.EXIT
+            const expectedRunwayModel = this.departureRunwayModel;
+            let entryPoint = airportIcao.toUpperCase() + expectedRunwayModel.name; // sidModel.getFirstEntryPoint();
+
+            if (!sidModel.hasEntry(entryPoint)) {
+                entryPoint = sidModel.getFirstEntryPoint();
+
+                if (_isEmpty(entryPoint)) {
+                    throw new TypeError(`the '${procedureId}' departure has no valid entry points`);
+                }
+            }
+
+            routeStringElements.unshift(entryPoint);
```

Comment:
we should really work to normalize casing with the routeString.  remember the old day's with `.toLowerCase()` and `.toUpperCase()` all over the place?  I don't want to go back to that.  Data comes in, we normalize to lower.  Data goes out, we transform to whats needed only then and never anywhere else in between.

```suggestion
            routeStringElements.unshift(entryPoint.toUpperCase());
```

Reply:
I agree, but the problems with that are widespread with the routes already. Should be addressed as part of #1173.